platform-sdk/swirlds-platform-core/src/test/java/com/swirlds/platform/sync/protocol/SyncProtocolTests.java

```diff
@@ -0,0 +1,485 @@
+/*
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.platform.sync.protocol;
+
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.Mockito.mock;
+
+import com.swirlds.common.system.NodeId;
+import com.swirlds.common.test.fixtures.FakeTime;
+import com.swirlds.common.threading.SyncPermitProvider;
+import com.swirlds.common.threading.locks.locked.MaybeLocked;
+import com.swirlds.platform.Connection;
+import com.swirlds.platform.components.CriticalQuorum;
+import com.swirlds.platform.metrics.SyncMetrics;
+import com.swirlds.platform.network.NetworkProtocolException;
+import com.swirlds.platform.sync.FallenBehindManager;
+import com.swirlds.platform.sync.ShadowGraphSynchronizer;
+import java.time.Duration;
+import java.util.List;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+
+/**
+ * Tests for {@link SyncProtocol}
+ */
+@DisplayName("Sync Protocol Tests")
+class SyncProtocolTests {
```

Comment:
Need to add a couple more tests:

1. Test that the sync permit is closed on `acceptFailed`
2. Test that the sync permit is closed even if the synchronizer throws an exception

Reply:
You can run test coverage easily in intelliJ to see what is covered and what is not. Let me know if you want me to show you how to do it.